source/common/event/dispatcher_impl.cc

```diff
@@ -119,9 +119,9 @@ DispatcherImpl::createSslListener(Network::ConnectionHandler& conn_handler,
 }
 
 TimerPtr DispatcherImpl::createTimer(TimerCb cb) {
-  // TODO(htuch): Figure out why we can't set this assertion, see
-  // https://github.com/lyft/envoy/issues/1294.
-  // ASSERT(isThreadSafe());
+  // We don't ASSERT that createTimer() is called on the run thread because timers are used for
```

Comment:
Would it make sense to have a `postAfter(some time, cb)` API instead, to make clear this is what is going on? While this may be the implementation, logically you aren't after creating some resource on another thread, you're after queueing up some post to be delivered after some time.